### ISSUE TITLE
Access violation in Rust 1.38

### ISSUE THREAD

**Issue by CoolOppo** [jessegrosjean/jwalk#11]
I made a new project to test this crate, here is my `main.rs`:

```rust
fn main() {
    use jwalk::WalkDir;

    for entry in WalkDir::new(".").sort(true) {
        println!("{}", entry.unwrap().path().display());
    }
}
```

On Windows, it returns `error: process didn't exit successfully: `target\release\test2.exe` (exit code: 0xc0000005, STATUS_ACCESS_VIOLATION)`

This only happens in a release build. In a debug build, it's fine.

**Comment by jessegrosjean** [comment]
I don't have windows to test on... generally that code seems to work on my MacOS machine in both debug and release mode... of course the content of `.` is maybe a deciding factor. What if you take out the forced `.unwrap()`?

**Comment by CoolOppo** [comment]
Still doesn't work.

```rust
use jwalk::WalkDir;

fn main() {
    for dir_entry in WalkDir::new(".").sort(true) {
        if let Ok(entry) = dir_entry {
            println!("{}", entry.path().display());
        }
    }
}
```

Same exact error. It also occurs on nightly (I am using `nightly-2019-10-04-x86_64-pc-windows-msvc`)

**Comment by CoolOppo** [comment]
I just ran it and for the first time got a different error message:

`(exit code: 0xc0000409, STATUS_STACK_BUFFER_OVERRUN)`

**Comment by CoolOppo** [comment]
If I set `num_threads` to `1`, everything works fine.

**Comment by jessegrosjean** [comment]
If you set num_threads to 1 then it takes a different codepath that doesn't use Rayon. Sorry I don't know why the Rayon path wouldn't work on windows, I've only tested macOS.

**Comment by jessegrosjean** [comment]
I just released 0.5 which fixes some issues with rayon and I "think" fixes this issue. Though I can't be sure since I couldn't recreate it. Please reopen if not fixed.
